src/EventBand/Transport/AmqpLib/AmqpLibDriver.php

```diff
@@ -129,7 +129,11 @@ function (AMQPLibMessage $msg) use (&$active, $callback, $channel, $queue) {
                     $error = error_get_last();
                     // Check if we got interruption from system call, ex. on signal
                     // On some php version (ex. php 5.5 error is null on interrupted system call)
-                    if ($error === null || stripos($error['message'], 'interrupted system call') !== false) {
+                    if (
+                        $error === null
+                        || stripos($error['message'], 'interrupted system call') !== false
+                        || (error_reporting() & $error['type']) === 0 // Don't consider errors not included in error reporting
```

Comment:
I think it's better to check that error is not `E_ERROR` or `E_WARNING`. User can define any `error_reporting` level but we should throw an exception anyway if something goes wrong in select